File: android/src/main/java/expo/modules/numerator/ExpoNumeratorModule.kt
package expo.modules.numerator

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.text.DecimalFormatSymbols
import java.util.Locale

class ExpoNumeratorModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoNumeratorModule")

    Function("getPlatformInfo") {
      mapOf(
        "platform" to "android",
        "moduleName" to "ExpoNumeratorModule",
        "native" to true
      )
    }

    Function("getPreferredLocale") {
      Locale.getDefault().toLanguageTag()
    }

    Function("getNumberSeparators") { localeTag: String? ->
      val locale = localeTag?.let { Locale.forLanguageTag(it) } ?: Locale.getDefault()
      val symbols = DecimalFormatSymbols.getInstance(locale)

      mapOf(
        "decimal" to symbols.decimalSeparator.toString(),
        "grouping" to symbols.groupingSeparator.toString()
      )
    }
  }
}

File: ios/ExpoNumeratorModule.swift
import ExpoModulesCore
import Foundation

public class ExpoNumeratorModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoNumeratorModule")

    Function("getPlatformInfo") {
      return [
        "platform": "ios",
        "moduleName": "ExpoNumeratorModule",
        "native": true
      ] as [String: Any]
    }

    Function("getPreferredLocale") {
      return Locale.autoupdatingCurrent.identifier.replacingOccurrences(of: "_", with: "-")
    }

    Function("getNumberSeparators") { (locale: String?) -> [String: String] in
      let formatter = NumberFormatter()
      formatter.locale = locale.map { Locale(identifier: $0) } ?? Locale.autoupdatingCurrent

      return [
        "decimal": formatter.decimalSeparator ?? ".",
        "grouping": formatter.groupingSeparator ?? ","
      ]
    }
  }
}
